fuseoverlayfs.go

```diff
@@ -182,9 +181,12 @@ func (o *snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, er
 	if err != nil {
 		return nil, err
 	}
+	defer t.Rollback()
 	s, err := storage.GetSnapshot(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get snapshot %s from storage: %w", key, err)
+	}
```

Comment:
Resolves ineffectual assignment of err. If `storage.GetSnapshot` call fails then `s == snapshots.Snapshot{}` or empty snapshot struct. From my read it made sense to propagate the error here instead of continuing and attempting to mount an empty snapshot. Lmk your thoughts.